extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -353,97 +320,86 @@ export abstract class AzureAuth implements vscode.Disposable {
+			Logger.error(ex);
+			throw new AzureAuthError(getMsg, 'Getting account from cache failed', ex);
+		}
 
-			try {
-				const tokenUrl = `${this.loginEndpointUrl}${tenant.id}/oauth2/token`;
-				const tokenResponse = await this.makePostRequest(tokenUrl, postData);
-				Logger.pii(JSON.stringify(tokenResponse.data));
-				const tokenClaims = this.getTokenClaims(tokenResponse.data.access_token);
+		try {
+			if (!accessTokenString) {
+				return undefined;
+			}
+			const accessToken: AccessToken = JSON.parse(accessTokenString);
+			let refreshToken: RefreshToken;
+			if (refreshTokenString) {
+				refreshToken = JSON.parse(refreshTokenString);
+			}
 
-				const accessToken: AccessToken = {
-					token: tokenResponse.data.access_token,
-					key: tokenClaims.oid ?? tokenClaims.email ?? tokenClaims.unique_name ?? tokenClaims.name,
-				};
+			return {
+				accessToken, refreshToken, expiresOn
+			};
+		} catch (ex) {
+			Logger.error(ex);
+			throw new AzureAuthError(parseMsg, 'Parsing account from cache failed', ex);
+		}
+	}
+	//#endregion
 
-				const refreshToken: RefreshToken = {
-					token: tokenResponse.data.refresh_token,
-					key: accessToken.key
-				};
-				const expiresOn = tokenResponse.data.expires_on;
-
-				refreshResponse = { accessToken, refreshToken, tokenClaims, expiresOn };
-			} catch (ex) {
-				Logger.pii(JSON.stringify(ex?.response?.data));
-				if (ex?.response?.data?.error === 'interaction_required') {
-					const shouldOpenLink = await this.openConsentDialog(tenant, resourceId);
-					if (shouldOpenLink === true) {
-						const { tokenRefreshResponse, authCompleteDeferred } = await this.promptForConsent(resourceEndpoint, tenant.id);
-						refreshResponse = tokenRefreshResponse;
-						authCompleteDeferred.resolve();
-					} else {
-						vscode.window.showInformationMessage(localize('azure.noConsentToReauth', "The authentication failed since Azure Data Studio was unable to open re-authentication page."));
-					}
-				} else {
-					return undefined;
-				}
+	//#region interaction handling
+
+	private async handleInteractionRequired(tenant: Tenant, resource: Resource): Promise<OAuthTokenResponse | undefined> {
+		const shouldOpen = await this.shouldOpenConsentDialog(tenant, resource);
+		if (shouldOpen) {
+			const result = await this.login(tenant, resource);
+			if (result && result.authComplete) {
+				result.authComplete.resolve();
 			}
 
-			this.memdb.set(this.createMemdbString(refreshResponse.accessToken.key, tenant.id, resourceId), refreshResponse.expiresOn);
-			return refreshResponse;
-		} catch (err) {
-			const msg = localize('azure.noToken', "Retrieving the Azure token failed. Please sign in again.");
-			vscode.window.showErrorMessage(msg);
-			throw new Error(err);
+			return result?.response;
 		}
+		return undefined;
 	}
 
-	public async openConsentDialog(tenant: Tenant, resourceId: string): Promise<boolean> {
+	private async shouldOpenConsentDialog(tenant: Tenant, resource: Resource) {
```

Comment:
Isn't this still opening the ConsentDialog?

And regardless - add return type

Reply:
Ah yeah, I guess I should've called it something else "shouldReauthenticate" would've been better.